src/core/tcp_main.c

```diff
@@ -985,16 +1196,29 @@ struct tcp_connection* tcpconn_new(int sock, union sockaddr_union* su,
 
 	atomic_set(&c->refcnt, 0);
 	local_timer_init(&c->timer, tcpconn_main_timeout, c, 0);
-	su2ip_addr(&c->rcv.src_ip, su);
-	c->rcv.src_port=su_getport(su);
-	c->rcv.bind_address=ba;
-	if (likely(local_addr)){
-		su2ip_addr(&c->rcv.dst_ip, local_addr);
-		c->rcv.dst_port=su_getport(local_addr);
-	}else if (ba){
-		c->rcv.dst_ip=ba->address;
-		c->rcv.dst_port=ba->port_no;
+	if (unlikely(ksr_tcp_accept_haproxy && state == S_CONN_ACCEPT)) {
+		ret = tcpconn_read_haproxy(c);
+
+		if (ret == -1) {
+			LM_ERR("invalid PROXY protocol header\n");
+			goto error;
+		} else if (ret == 1) {
+			LM_DBG("PROXY protocol did not override IP addresses\n");
+			goto read_ip_info;
+		}
```

Comment:
Small suggestion, add a LM_DBG output here, that the tcpconn_read_haproxy call was successful.

Reply:
There is an `LM_DBG()` indicating whether we received a PROXY v1 or PROXY v2 header. Do you feel that a parsing-success message is strictly required here (as opposed to a lack of parsing-failure message)?